### PR TITLE
BakeHelper.php optionize indent string

### PR DESCRIPTION
**Pull request by schrolli**
Make the indent-string an option to give users, who want to use their own Theme, the ability to justify the indentation, e.g. indent by tab instead of four spaces.

### CODE REVIEW

**Comment by dereuromark** [comment]
:+1: Nice!

**Comment by schrolli** [comment]
what are these checks? did i made an error?

**Comment by markstory** [comment]
@schrolli The checks are from the various CI services we use. AppVeyor failed because of heap corruption. And the postgres tests have been failing for a while and I don't know why. I don't think your changes caused either of these builds to fail directly.

**Comment by schrolli** [comment]
ah, ok. was just wondering;)

**Comment by markstory** [comment]
@schrolli Could you add a test case for this? Having that would help ensure we don't accidentally break this feature in the future.

**Comment by schrolli** [comment]
i added an another option and added some testcases to cover pretty much all the options

File: src/View/Helper/BakeHelper.php [hunk 1]
```diff
@@ -62,7 +62,9 @@ public function arrayProperty($name, array $value = [], array $options = [])
     public function stringifyList(array $list, array $options = [])
     {
         $options += [
-            'indent' => 2
+            'indent' => 2,
+            'tab' => '    ',
+            'comma_at_end' => false,
```
Comment (markstory) [inline src/View/Helper/BakeHelper.php]: Should be commaAtEnd.

File: src/View/Helper/BakeHelper.php [hunk 2]
```diff
@@ -80,9 +82,13 @@ public function stringifyList(array $list, array $options = [])
...
             $join = ',';
-            $start = "\n" . str_repeat("    ", $options['indent']);
+            $start = "\n" . str_repeat($options['tab'], $options['indent']);
             $join .= $start;
-            $end = "\n" . str_repeat("    ", $options['indent'] - 1);
+            $end = "\n" . str_repeat($options['tab'], $options['indent'] - 1);
+        }
+        
+        if ($options['comma_at_end']) {
+            $end = ",".$end;
```
Comment (markstory) [inline src/View/Helper/BakeHelper.php]: Spaces around operators.

File: tests/TestCase/View/Helper/BakeHelperTest.php [hunk 3]
```diff
@@ -119,4 +119,105 @@ public function testAliasExtractorBelongsToMany()
...
+     */
+    public function testStringifyListDefaults()
+    {
+        $list = ['one' => 'foo', 'two' => 'bar', 'three'];
+        $result = $this->BakeHelper->stringifyList($list);
+        $spaces = '    ';
+        $expected = "\n".
+            $spaces.$spaces."'one' => 'foo',\n".
+            $spaces.$spaces."'two' => 'bar'\n".
+            $spaces.$spaces."'three'\n".
```
Comment (markstory) [inline tests/TestCase/View/Helper/BakeHelperTest.php]: Spaces around operators improve readability and consistency with the rest of the project.

**Comment by schrolli** [comment]
oh, i wasn't aware of concatenations and psr-2... now all dots should have spaces and the option is renamed;)

**Comment by markstory** [comment]
Looking better, but [some of the tests are failing](https://travis-ci.org/cakephp/bake/jobs/81888566)

**Comment by schrolli** [comment]
bump

**Comment by AD7six** [comment]
@schrolli There are still phpcs fails - click on the red x.

**Comment by schrolli** [comment]
ah, now the ci-job runs through, hadn't saw the error in the comments of the script...

File: src/View/Helper/BakeHelper.php [hunk 4]
```diff
@@ -62,7 +62,9 @@ public function arrayProperty($name, array $value = [], array $options = [])
     public function stringifyList(array $list, array $options = [])
     {
         $options += [
-            'indent' => 2
+            'indent' => 2,
+            'tab' => '    ',
+            'commaAtEnd' => false,
```
Comment (markstory) [inline src/View/Helper/BakeHelper.php]: `trailingComma` might be a better name for this.

**Comment by schrolli** [comment]
so the appveyor-error is related to the core with some sql-errors and i can't do something about it...

the travis-ci test is passing

what i do not understand: the decreased coverage at coveralls. The decrease is located at the end of the default-config array. has it possibly something to do with the trailing comma in the line before? but is a trailing comma bad practice then?

**Comment by dereuromark** [comment]
I wouldn't wry about any one of those two. Looks all good.

Personally, I think trailing comma can be a good practice if enforced automatically via cs (not manually), as it helps to keep merge diffs to a minimum without additional efforts for the developer. But that is out of scope of the actual PR here :P

**Pull request merged**